packages/api/src/utils/pin.js

```diff
@@ -139,8 +139,6 @@ export async function waitOkPins (cid, cluster, waitTime = MAX_PIN_STATUS_CHECK_
  */
 export async function waitAndUpdateOkPins (cid, cluster, db, waitTime = MAX_PIN_STATUS_CHECK_TIME, checkInterval = PIN_STATUS_CHECK_INTERVAL) {
   const okPins = await waitOkPins(cid, cluster, waitTime, checkInterval)
-  for (const pin of okPins) {
-    await db.upsertPin(cid, pin)
-  }
+  await db.upsertPins(okPins)
```

Comment:
I think there's something going on with passing the wrong objects here.

If you compare [upserPin](https://github.com/web3-storage/web3.storage/blob/1373dbbffa804b04f2f23d80d142fc960d28af6c/packages/db/index.js#L419) and [upsertPins](https://github.com/web3-storage/web3.storage/blob/main/packages/db/index.js#L447) they are quite different.

[upserPin](https://github.com/web3-storage/web3.storage/blob/1373dbbffa804b04f2f23d80d142fc960d28af6c/packages/db/index.js#L419) call a remote procedure and takes care of creating/updating the location in the DB while [upsertPins](https://github.com/web3-storage/web3.storage/blob/main/packages/db/index.js#L447) is a straight upsert in the `pin` table.

The problem here is also that the parameter type annotation is wrong [here](https://github.com/web3-storage/web3.storage/blob/1373dbbffa804b04f2f23d80d142fc960d28af6c/packages/db/index.js#L445), the function does not expect an `PinUpsertInput[]`, especially if you look at the item shape of that array, it should have a `pin_location_id: string` instead of a `location: Location`. 

Please have a look [here](https://github.com/web3-storage/web3.storage/blob/1373dbbffa804b04f2f23d80d142fc960d28af6c/packages/cron/src/jobs/pins.js#L101) to have an example of what is actually expected. 

We need to make sure we update the function type annotation and we pass the right objects around.

Reply:
Created a new type `PinsUpsertInput` and used this instead.